src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateString.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Core
+{
+    /// <summary>
+    /// Represents a string in template.json file that needs to be localized.
+    /// </summary>
+    internal struct TemplateString : IEquatable<TemplateString>
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="TemplateString"/>.
+        /// </summary>
+        /// <param name="identifier">A string that uniquely identifies this template field in the template.json file.</param>
+        /// <param name="key">A string that uniquely identifies this field in localized templatestrings.json file.</param>
+        /// <param name="value">Localizable string.</param>
+        public TemplateString(string identifier, string key, string value)
+        {
+            Identifier = identifier;
+            Key = key;
+            Value = value;
+        }
+
+        /// <summary>
+        /// Gets the string that identifies this localizable string in template.json file.
+        /// Given a template.json file and an <see cref="Identifier"/>, it is possible to
+        /// directly find the corresponding JSON element.
+        /// </summary>
+        /// <example>"symbols.Framework.choices[0].displayName".</example>
+        public string Identifier { get; }
+
+        /// <summary>
+        /// Gets the key of this localizable string, which identifies this in templatestrings.json
+        /// file. This is different than <see cref="Identifier"/> in the way that it is intended to
+        /// be more user friendly.
+        /// </summary>
+        public string Key { get; }
```

Comment:
why it has to be different form `Identifier`? can we do the extension for `Identifier` that converts `Identifier` to `Key`? 
maybe example for `symbols.Framework.choices[0].displayName` will make it a bit easier to understand

Reply:
We have resolved this after internal discussions.